### ISSUE TITLE
Won't convert .avsc to .avdl when there are referenced schema in the source .avsc

### ISSUE THREAD

**Issue by q3769-patientpoint** [opwvhk/avro-schema-support#167]
**Description**

Won't convert .avsc to .avdl when there are referenced schema (referenced file in same directory) in the source .avsc. If there is no referenced schema, then it seems to work fine.

**To Reproduce**

Steps to reproduce the behaviour:
1. Start with a valid .avdl file (e.g. one generated from an existing/valid .avsc file)
2. Try Refactor -> Convert Avro Schema to Avro IDL

**Expected behaviour**

A popup window asking for "Save". Once confirmed, will save newly generated .avdl file.

**Screenshots**

No popup window, just says "parsing" and no further action.

![image](https://github.com/user-attachments/assets/9e1991d6-b840-410f-b6e9-bff3bb39c860)

**Versions:**
IntelliJ IDEA 2024.2.0.2 (Ultimate Edition)
Build #IU-242.20224.419, built on August 19, 2024
 - Plugin: Version 223.3.1

**Additional context**

If there is no referenced schema, and all definitions are inside the same source .avsc file, then it seems to work fine.

**Comment by opwvhk** [comment]
Thank you for this detailed bug report!

**TL;DR:** This is at partially a bug, and partially missing/unsupported functionality. The workaround: select all files that contain a complete set of schemata.

**Longer explanation:**

This issue is related to / a duplicate of the crash report #162: the parsing process fails because not all schemata are defined. The way the plugin doesn't do anything is a bug.

On the other hand, the failure to create a `.avdl` file is expected: as `.avdl` files allow imports, they are assumed to hold a complete set of schemata. This in contrast to `.avsc`/`.avpr` files, which cannot define imports. The Maven plugin, for example, allows to parse a preliminary set of schema files as "imports" to overcome this limitation.

The current `.avdl` file generation is not capable of figuring out which files to import. As a result, you must do this manually. The resulting file defines all schemata in all selected files.
